src/main/java/seedu/duke/Parser.java

```diff
@@ -0,0 +1,135 @@
+package main.java.seedu.duke;
+
+import java.time.LocalDate;
+import java.time.format.DateTimeFormatter;
+import java.util.regex.Pattern;
+
+import main.java.seedu.duke.commands.*;
+import main.java.seedu.duke.todo.Deadline;
+import main.java.seedu.duke.todo.Event;
+import main.java.seedu.duke.todo.Todo;
+import main.java.seedu.duke.todo.Task;
+
+/**
+ * Used for parsing strings for Duke project.
+ */
+public class Parser {
+    private static DateTimeFormatter formatter = DateTimeFormatter.ofPattern("MMM d yyyy");
+
+    /**
+     * Returns corresponding command specified in the input string.
+     * @param str The input string.
+     * @return Command.
+     * @throws DukeException If the input is incomplete.
+     */
+    public static Command parse(String str) throws DukeException {
```

Comment:
Perhaps consider a clearer name for the method's parameter such as "command" or "input"?